resources/RegexGrammar.pp

```diff
@@ -132,7 +135,7 @@
     alternation()
 
 alternation:
-    concatenation() ( ::alternation:: concatenation() #alternation )*
+    concatenation()? ( <alternation> concatenation()? #alternation )*
```

Comment:
@staabm would you sending a PR into this PR branch for this change?

I looked into https://github.com/phpstan/phpstan-src/blob/1.12.5/src/Type/Regex/RegexGroupParser.php#L180 and I do not know the internals, why count is passed etc.

The effective change of this line is `|` tokens are newly captured inside the AST and if there are empty tokens before/after `|`, then the alternation expression matches `''`.

Reply:
sorry, I don't get what you ask me for...?

Reply:
Previously "PCRE alternation" captured only the tokens between `|`, newly also `|` tokens are captured and the parsing code needs to be adjusted.

Reply:
Well, I coped with tha by modifying the parsed AST to produce the same AST as before this PR but with empty tokens added.